util/sparse/punch_windows.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build windows
+
+package sparse
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// punchAt for Windows also marks the file as sparse before it punches a hole in it.
```

Comment:
It'd be nice to point to the documentation so that we don't have to google it every time.